### ISSUE TITLE
SCRAM-SHA-1 missing authSource?

### ISSUE THREAD

**Issue by colinbes** [mafintosh/mongojs#222]
In latest update I see mention of passing an option for specifying authMechanism.

```
In mongojs:
mongojs('username:password@localhost/bcdb', ['mycollection'], {authMechanism: 'ScramSHA1'});
```

but I don't see how to specify the authority source database for authentication?

For example, in mongodb it's defined in the url along with the authentication mechanism.

```
mongodb, mongoose etc:
"mongodb://username:password@localhost:27017/bcdb/?authSource=bcdb&authMechanism=SCRAM-SHA-1"
```

More than happy to help with any testing/validation etc.

Thanks
Colin

**Comment by saintedlama** [comment]
Did take a look mongodb-core that mongodb driver mongojs uses under the hood and found nothing to specify authSource. Already created an issue in mongodb-core project https://github.com/christkv/mongodb-core/issues/32

**Comment by saintedlama** [comment]
This has to be implemented in mongojs since mongodb-core is a bit to lean for this. Any help appreciated!

**Comment by colinbes** [comment]
See https://patch-diff.githubusercontent.com/raw/mafintosh/mongojs/pull/231.patch as a quick fix I tried.

**Comment by saintedlama** [comment]
Should be fixed in version 2.0 (released just some seconds ago)

**Comment by colinbes** [comment]
Sweet, thanks!
